doc/contrib/recording.md

```diff
@@ -0,0 +1,81 @@
+# Session Recording Technical Documentation
+
+After every `kdbSet` action, the changes are calculated using Elektra's powerful changetracking API.
+The result of the calculation is an `ElektraDiff` instance we'll call _part diff_ throughout this document.
+The session recording plugin merges those _part diffs_ together and creates and persists an overall _session diff_.
+
+Conceptually, this is depicted in the following image:
+
+![Session vs Part diff](../images/elektra-record/recording-part-diff.svg)
+
+Importantly, the _session diff_ is shared across processes.
+The _session diff_ persistently records all changes made to the whole KDB from any process between the start and end of the session.
+Because of the cross-process nature of a _session diff_, it is also susceptible to "simultaneous write" conflicts.
+We tried to limit the impact of these conflicts through locking.
+
+All persistable namespaces are monitored for changes.
+
+We currently have no way of recording changes done outside of Elektra, i.e. when the configuration files got edited manually.
+
+## Storage of the Session Diff
+
+The session diff is persisted in the respective namespace under `<namespace>:/elektra/record/session`.
+I.e. all keys in the diff of the `system` namespace are under `system:/elektra/record/session`.
+
+The recording plugin needs its own KDB instance to store the session diff within Elektra.
+For concurrency and performance reasons, the storage should be a separately mounted file under each namespace.
+This means we provide hard coded default mountpoints for the
+
+- `dir:/elektra/record/session`,
+- `system:/elektra/record/session`,
+- `spec:/elektra/record/session` and
+- `user:/elektra/record/session`
+  keys.
+
+These mountpoints store the keys in the `dump` format in a file called `record-session.cfg` in the respective standard directories for their namespace.
```

Comment:
I don't think we should hard-code dump here. quickdump, mmapstorage are probably more suited in many use cases. What about a symlink like we already have for default storage?

Reply:
Well it uses the same code as the mounting of default storage. So I probably should just update this sentence to reflect this.